sdk/cxx/include/sawtooth/transaction_handler.h

```diff
@@ -19,98 +19,113 @@
 #include <list>
 #include <string>
 
+#include "sawtooth_sdk.h"
 #include "proto/transaction.pb.h"
 
 #include "sawtooth/global_state.h"
 
 namespace sawtooth {
 
-// Utility types to allow for managed instances of std::string on the heap.
-typedef std::unique_ptr<std::string> StringUPtr;
-typedef std::shared_ptr<std::string> StringPtr;
-
-// The transaction data for a Transaction Processing request.
-class Transaction final {
- public:
-    Transaction(TransactionHeader* header, StringPtr payload, StringPtr signature):
-            payload_(payload), signature_(signature) {
-    }
-
-    Transaction (const Transaction&) = delete;
-    Transaction (const Transaction&&) = delete;
-    Transaction& operator= (const Transaction&) = delete;
-
-    const TransactionHeader& header() const {
-        return this->header_;
-    }
-
-    const std::string& payload() const {
-        return *(this->payload_);
-    }
-
-    const std::string& signature() const {
-        return *(this->signature_);
-    }
-
- private:
-    TransactionHeader header_;
-    StringPtr payload_;
-    StringPtr signature_;
+class TxHeaderWrapper: public TxHeaderIF
```

Comment:
We do not use 'Tx' to mean Transaction anywhere else in the code, and generally frown on abbreviations.  The C++ SDK should generally be made to look closer to the other SDKs.

Reply:
I can change Tx to Transaction, but curios why Tx is not welcomed. It makes variables shorter

Reply:
The reason for avoiding abbreviations in general is for code clarity, both for English and non-native-English speakers. Abbreviations tend to cause the reader to require more information.  This is obviously subjective but generally the rule we use.  There are exceptions too, of course, especially when we are declaring local variables, etc.  But specifically here, I think we want consistency with the rest of the SDKs to the extent that makes sense.